templates/master/01-master-kubelet/_base/units/kubelet.service.yaml

```diff
@@ -4,7 +4,7 @@ contents: |
   [Unit]
   Description=Kubernetes Kubelet
   Wants=rpc-statd.service network-online.target crio.service
-  After=network-online.target crio.service
+  After=network-online.target crio.service systemd-hostnamed.service
```

Comment:
Does this fix the bug? Unless the hostname is statically set, then you won't see an `/etc/hostname`. A lot of the platforms use a transient hostname. 

AFAIK, mounting in `/proc/kernel/hostname` would be better since it would have the current hostname.

Reply:
hmmm I hoped that systemd-hostnamed finishing would mean *something* would be in /etc/hostname.

That's a good idea! I will look into making that switch

Reply:
I cannot find what container actually attempts to mount /etc/hostname. The only thing cri-o can really do is switch references to it to /proc/sys/kernel/hostname, which doesn't seem that safe...